src/pino_transport.ts

```diff
@@ -0,0 +1,100 @@
+import build from "pino-abstract-transport"
+import { Client } from "./client"
+
+type PinoTransportOptions = {
+  client: Client
+  group: string
+}
+
+const appsignalPinoTransport = ({
+  client,
+  group = "app"
+}: PinoTransportOptions) => {
+  let buffer: Record<string, any>[] = []
+  let timeout: NodeJS.Timeout | null = null
+
+  return build(
+    async (source: any) => {
+      for await (const obj of source) {
+        buffer.push(parseInfo(obj, group))
+
+        if (!timeout) {
+          timeout = setTimeout(flush, 1000)
+        }
+      }
+    },
+    { close: flush }
+  )
+
+  async function flush() {
+    if (timeout) {
+      clearTimeout(timeout)
+    }
+    timeout = null
+
+    if (buffer) {
+      const data = buffer
+      buffer = []
+      data.forEach(line => sendLogs(line, client))
+    }
+  }
+
+  async function sendLogs(data: Record<string, any>, client: Client) {
+    client.extension.log(
+      data.group || "app",
+      data.severity,
+      0,
+      data.msg,
+      data.attributes
+    )
+  }
+}
+
+function parseInfo(
+  obj: Record<string, any>,
+  group: string
+): Record<string, any> {
+  const { hostname, level, msg, ...attributes } = obj
+
+  return {
+    severity: getSeverity(level),
+    hostname,
+    group,
+    msg,
+    attributes: flattenAttributes(attributes)
+  }
+}
+
+function flattenAttributes(
+  attributes: Record<string, any>,
+  prefix = ""
+): Record<string, any> {
+  let result: Record<string, any> = {}
+
+  for (const key in attributes) {
+    const newKey = prefix ? `${prefix}.${key}` : key
+
+    if (
+      typeof attributes[key] === "object" &&
+      attributes[key] !== null &&
+      !Array.isArray(attributes[key])
+    ) {
+      const flattened = flattenAttributes(attributes[key], newKey)
+      result = { ...result, ...flattened }
+    } else {
+      result[newKey] = attributes[key]
+    }
+  }
+
+  return result
+}
+
+function getSeverity(level: number): number {
+  if (level >= 50) return 6
+  if (level >= 40) return 5
+  if (level >= 30) return 3
+  if (level >= 20) return 2
+  return 1
```

Comment:
It might help for readability to refer to our [logger level constants](https://github.com/appsignal/appsignal-nodejs/blob/main/src/logger.ts#L7-L14) -- so, instead of `return 6`, `return LOGGER_LEVEL_SEVERITY.error`, and so on.